### PR TITLE
docs: translate to Japanese tests.md

### PR DESCRIPTION
**Pull request by tkoyama010**
ref #352

### CODE REVIEW

File: locales/ja/LC_MESSAGES/tests.po [hunk 1]
```diff
@@ -22,71 +22,79 @@ msgstr ""
...
 " for your Python package and how you can setup infrastructure to run your"
 " tests both locally and on GitHub."
 msgstr ""
+"このセクションでは、Python パッケージのテストを書くことの重要性と、ローカルと GitHub "
+"の両方でテストを実行するためのインフラストラクチャの設定方法について学びます。"
 
 #: ../../tests/index.md:19
-msgid "✨ Why write tests ✨"
-msgstr ""
+msgid "笨ｨ Why write tests 笨ｨ"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:70]:
```suggestion
msgid "✨ Why write tests ✨"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 2]
```diff
@@ -22,71 +22,79 @@ msgstr ""
...
 "about why you should write tests and how they can help you and potential "
 "contributors to your project."
 msgstr ""
+"Python パッケージのテストを書く技術についてもっと学びましょう。 "
+"なぜテストを書く必要があるのか、そしてテストがどのようにあなたやあなたのプロジェクトへの潜在的な貢献者の助けになるのかを学びましょう。"
 
 #: ../../tests/index.md:31
-msgid "✨ Types of tests ✨"
-msgstr ""
+msgid "笨ｨ Types of tests 笨ｨ"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:83]:
```suggestion
msgid "✨ Types of tests ✨"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 3]
```diff
@@ -22,71 +22,79 @@ msgstr ""
...
 " package: unit tests, integration tests and end-to-end (or functional) "
 "tests. Learn about all three."
 msgstr ""
+"Python パッケージで書くことのできるテストには、ユニットテスト、統合テスト、エンドツーエンドテスト (あるいは機能テスト) の 3 "
+"種類があります。 この 3 つについて学びましょう。"
 
 #: ../../tests/index.md:42
-msgid "✨ Run tests locally ✨"
-msgstr ""
+msgid "笨ｨ Run tests locally 笨ｨ"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:96]:
```suggestion
msgid "✨ Run tests locally ✨"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 4]
```diff
@@ -95,10 +103,13 @@ msgid ""
...
 "across python versions here."
 msgstr ""
+"もしユーザがあなたのパッケージを異なるバージョンの Python で使うことを想定しているなら、テストを実行するために nox "
+"のような自動化ツールを使うと便利です。 Python "
+"のバージョンにまたがってテストを実行するために使える様々なツールについては、こちらを参照してください。"
 
 #: ../../tests/index.md:53
-msgid "✨ Run tests online (using CI) ✨"
-msgstr ""
+msgid "笨ｨ Run tests online (using CI) 笨ｨ"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:111]:
```suggestion
msgid "✨ Run tests online (using CI) ✨"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 5]
```diff
@@ -239,94 +276,106 @@ msgstr ""
...
 #: ../../tests/run-tests.md:83
 msgid "Learn more about code coverage here. (add link)"
-msgstr ""
+msgstr "コードカバレッジについてはこちらをご覧ください。 (リンクを追加)"
 
 #: ../../tests/run-tests.md:87
 msgid ""
 "Your editor or IDE may add additional convenience for running tests, "
-"setting breakpoints, and toggling the `–no-cov` flag. Check your editor's"
+"setting breakpoints, and toggling the `窶渡o-cov` flag. Check your editor's"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:343]:
```suggestion
"setting breakpoints, and toggling the `–no-cov` flag. Check your editor's"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 6]
```diff
@@ -606,15 +708,17 @@ msgid ""
...
 "method level, performs its intended task correctly."
 msgstr ""
+"単体テストは、個々のコンポーネントやコードの単位を分離してテストし、それらが正しく動作することを確認するものです。単体テストの目的は、ソフトウェアの各部分、通常は関数やメソッドのレベルが、意図したタスクを正しく実行することを検証することです。"
 
 #: ../../tests/test-types.md:24
 msgid ""
 "Unit tests can be compared to examining each piece of your puzzle to "
 "ensure parts of it are not broken. If all of the pieces of your puzzle "
-"don’t fit together, you will never complete it. Similarly, when working "
+"don窶冲 fit together, you will never complete it. Similarly, when working "
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:717]:
```suggestion
"don’t fit together, you will never complete it. Similarly, when working "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 7]
```diff
@@ -663,31 +770,33 @@ msgid ""
 "integration test would ensure that all parts of that workflow behaved as "
 "expected."
 msgstr ""
+"例えば、スプレッドシートに温度データを収集し、摂氏から華氏に変換し、特定の期間の平均温度を提供する一連のステップがあったとします。統合テストは、そのワークフローのすべての部分が期待通りに動作することを保証します。"
 
 #: ../../tests/test-types.md:107
 msgid ""
 "image of two puzzle pieces with some missing parts. The puzzle pieces are"
-" purple teal yellow and blue. The shapes of each piece don’t fit "
+" purple teal yellow and blue. The shapes of each piece don窶冲 fit "
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:778]:
```suggestion
" purple teal yellow and blue. The shapes of each piece don窶冲 fit "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 8]
```diff
@@ -663,31 +770,33 @@ msgid ""
...
 "together."
-msgstr ""
+msgstr "パズルのピースが2つあり、一部欠けている画像です。パズルのピースは紫・鴨の羽色・黄色・青色。 それぞれのピースの形が合いません。"
 
 #: ../../tests/test-types.md:112
 msgid ""
-"If puzzle pieces have missing ends, they can’t work together with other "
+"If puzzle pieces have missing ends, they can窶冲 work together with other "
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:784]:
```suggestion
"If puzzle pieces have missing ends, they can’t work together with other "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 9]
```diff
@@ -932,42 +1089,47 @@ msgid ""
...
 
 #: ../../tests/write-tests.md:44
 msgid "Test examples"
-msgstr ""
+msgstr "テスト例"
 
 #: ../../tests/write-tests.md:47
 msgid ""
-"Let’s say you have a Python function that adds two numbers a and b "
+"Let窶冱 say you have a Python function that adds two numbers a and b "
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:1102]:
```suggestion
"Let’s say you have a Python function that adds two numbers a and b "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 10]
```diff
@@ -932,42 +1089,47 @@ msgid ""
...
 msgid ""
 "A test to ensure that function runs as you might expect when provided "
 "with different numbers might look like this:"
-msgstr ""
+msgstr "異なる数値が与えられたときに、その関数が期待通りに実行されることを確認するテストは次のようになります:"
 
 #: ../../tests/write-tests.md:72
-msgid "🧩🐍"
-msgstr ""
+msgid "ｧｩ錐"
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:1113]:
```suggestion
msgid "🧩🐍"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 11]
```diff
@@ -663,13 +770,14 @@ msgid ""
 "integration test would ensure that all parts of that workflow behaved as "
 "expected."
 msgstr ""
+"例えば、スプレッドシートに温度データを収集し、摂氏から華氏に変換し、特定の期間の平均温度を提供する一連のステップがあったとします。統合テストは、そのワークフローのすべての部分が期待通りに動作することを保証します。"
 
 #: ../../tests/test-types.md:107
 msgid ""
 "image of two puzzle pieces with some missing parts. The puzzle pieces are"
-" purple teal yellow and blue. The shapes of each piece don’t fit "
+" purple teal yellow and blue. The shapes of each piece don窶冲 fit "
```
Comment (tkoyama010) [inline locales/ja/LC_MESSAGES/tests.po:778]:
```suggestion
" purple teal yellow and blue. The shapes of each piece don’t fit "
```

**Comment by tkoyama010** [comment]
@kozo2 Same with https://github.com/pyOpenSci/python-package-guide/pull/367#issuecomment-2282335731

File: locales/ja/LC_MESSAGES/tests.po [hunk 12]
```diff
@@ -519,85 +609,97 @@ msgid ""
...
 msgid "End-to-end (also known as Functional) tests"
-msgstr ""
+msgstr "エンドツーエンド(機能テストとも呼ばれる)テスト"
 
 #: ../../tests/test-types.md:12
 msgid ""
 "Each type of test has a different purpose. Here, you will learn about all"
 " three types of tests."
-msgstr ""
+msgstr "それぞれの検査には異なる目的がある。 ここでは、3種類の検査すべてについて学びます。"
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:690]:
```suggestion
msgstr "それぞれの検査には異なる目的があります。 ここでは、3種類の検査すべてについて学びます。"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 13]
```diff
@@ -803,26 +926,32 @@ msgid ""
 " more without having to create development environments, run tests and "
 "build documentation locally."
 msgstr ""
+"CIは、外部の人があなたのソフトウェアに貢献している場合、非常に貴重です。 "
+"リポジトリに投稿されたすべてのプルリクエストに対してCIを実行するように設定できます。 "
+"CIは、あなたのリポジトリを新しい潜在的な貢献者によりフレンドリーなものにします。 "
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:931]:
```suggestion
"CIは、あなたのリポジトリを新しい将来の貢献者によりフレンドリーなものにします。 "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 14]
```diff
@@ -844,24 +975,31 @@ msgid ""
 " code alterations from contributors won't breaking existing "
 "functionality."
 msgstr ""
+"テストはコード変更のセーフティネットとして機能する。 バグがユーザーに影響を与える前に発見し、修正するのに役立ちます。 "
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:978]:
```suggestion
"テストはコード変更のセーフティネットとして機能します。 バグがユーザーに影響を与える前に発見し、修正するのに役立ちます。 "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 15]
```diff
@@ -844,24 +975,31 @@ msgid ""
...
+"テストはまた、貢献者によるコード改変が既存の機能を壊さないという確信を与えます。"
 
 #: ../../tests/write-tests.md:13
 msgid "Writing tests for your Python package is important because:"
-msgstr ""
+msgstr "Pythonパッケージのテストを書くことは重要である:"
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:983]:
```suggestion
msgstr "Pythonパッケージのテストを書くことは重要です:"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 16]
```diff
@@ -901,10 +1048,13 @@ msgid ""
 "your code base. This ensures that your software behaves the way you "
 "expect it to."
 msgstr ""
+"**より簡単なプルリクエストのレビュー:** GitHub アクションのような CI "
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:1051]:
```suggestion
"**より簡単なプルリクエストのレビュー:** GitHub actions のような CI "
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 17]
```diff
@@ -976,13 +1138,17 @@ msgid ""
 " to add two numbers, test that the outcome value of adding those two "
 "numbers is correct."
 msgstr ""
+"**いくつかの典型的なケースをテスト:** "
+"パッケージが、ユーザーが使用したときに期待通りに機能することをテストします。例えば、パッケージが2つの数値を足すことになっている場合、その2つの数値を足した結果が正しいかどうかをテストする。"
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:1142]:
```suggestion
"パッケージが、ユーザーが使用したときに期待通りに機能することをテストします。例えば、パッケージが2つの数値を足すことになっている場合、その2つの数値を足した結果が正しいかどうかをテストします。"
```

File: locales/ja/LC_MESSAGES/tests.po [hunk 18]
```diff
@@ -1000,3 +1168,5 @@ msgid ""
 "the user can easily understand why if failed (provides a useful error "
 "message)."
 msgstr ""
+"**コードが正しく失敗するかテストする** 関数が1以上の値を必要とする場合は、0.999でテストします。 "
+"予期しない値やヘルプが与えられたときに、関数が優雅に失敗し、ユーザが失敗した理由を簡単に理解できるようにします(有用なエラーメッセージを提供します)。"
```
Comment (kozo2) [inline locales/ja/LC_MESSAGES/tests.po:1172]:
```suggestion
"予期しない値やヘルプが与えられたときに、関数が潔く事を荒立てずに失敗し、ユーザが失敗した理由を簡単に理解できるようにします(有用なエラーメッセージを提供します)。"
```

**Review by kozo2 — COMMENTED** [review]
I finished my review.

File: locales/ja/LC_MESSAGES/tests.po (hunk 18 above)
Comment (tkoyama010) [reply locales/ja/LC_MESSAGES/tests.po:1172]: "gracefully" also means "正常に". How about this?
```suggestion
"予期しない値やヘルプが与えられたときに、関数が正常に失敗し、ユーザが失敗した理由を簡単に理解できるようにします(有用なエラーメッセージを提供します)。"
```

Comment (kozo2) [reply locales/ja/LC_MESSAGES/tests.po:1172]: "gracefully" also means "差し障り無く". How about this?
```suggestion
"予期しない値やヘルプが与えられたときに、関数が差し障り無く失敗し、ユーザが失敗した理由を簡単に理解できるようにします(有用なエラーメッセージを提供します)。"
```

**Comment by tkoyama010** [comment]
@kozo2 Thanks for reviewing. Your review is always perfect!

**Review by willingc — APPROVED** [review]
Thanks @tkoyama010

**Pull request merged**